sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/CertificateValidationCheckBase.cs

```diff
@@ -0,0 +1,298 @@
+                    {
+                        c.VisitedMethods.Add(syntax);
+                        return InvocationLocations(c, syntax);
+                    }
+                    break;
+            }
+            return ImmutableArray<Location>.Empty;
+        }
+
+        private ImmutableArray<Location> IdentifierLocations(InspectionContext c, SyntaxNode syntax)
+        {
+            switch (syntax)
+            {
+                case TMethodSyntax method:                  //Direct delegate name
+                    return BlockLocations(c, method);  
+                case TParameterSyntax parameter:            //Value arrived as a parameter
+                    return ParamLocations(c, parameter);
+                case TVariableSyntax variable:     //Value passed as variable
+                    return VariableLocations(c, variable);
+            }
+            return ImmutableArray<Location>.Empty;
+        }
+
+        protected ImmutableArray<Location> ParamLocations(InspectionContext c, TParameterSyntax param)
+        {
+            var ret = ImmutableArray.CreateBuilder<Location>();
+            var containingMethodDeclaration = param.FirstAncestorOrSelf<TMethodSyntax>();
+            if (!c.VisitedMethods.Contains(containingMethodDeclaration))
+            {
+                c.VisitedMethods.Add(containingMethodDeclaration);
+                var containingMethod = c.Context.SemanticModel.GetDeclaredSymbol(containingMethodDeclaration) as IMethodSymbol;
+                var paramSymbol = containingMethod.Parameters.Single(x => x.Name == IdentifierText(param));
+                foreach (var invocation in FindInvocationList(c.Context, FindRootClassOrModule(param), containingMethod))
+                {
+                    var methodParamLookup = CreateParameterLookup(invocation, containingMethod);
+                    if (methodParamLookup.TryGetSymbolParameter(paramSymbol, out var argument))
+                    {
+                        ret.AddRange(CallStackSublocations(c, ArgumentExpression(argument)));
+                    }
+                }
+            }
+            return ret.ToImmutable();
+        }
+
+        private ImmutableArray<Location> VariableLocations(InspectionContext c, TVariableSyntax variable)
+        {
+            var allAssignedExpressions = new List<TExpressionSyntax>();
+            var parentScope = LocalVariableScope(variable);
+            if (parentScope != null)
+            {
+                allAssignedExpressions.AddRange(parentScope.DescendantNodes().OfType<TAssignmentExpressionSyntax>()
+                    .Select(x =>
+                    {
+                        SplitAssignment(x, out var leftIdentifier, out var right);
+                        return new { leftIdentifier, right };
+                    })
+                    .Where( x => x.leftIdentifier != null && IdentifierText(x.leftIdentifier) == IdentifierText(variable))
```

Comment:
You can extract `IdentifierText(variable)` outside the iteration as it will always have the same value